src/navigation/TabNavigator.tsx

```diff
@@ -28,26 +33,69 @@ const TabNavigator = () => {
     >
       <Tab.Screen
         name="HomeTab"
-        component={StackNavigator}
+        component={HomeStackNavigator}
         options={({ route }) => ({
           tabBarLabel: t('Home')!,
           tabBarIcon: ({ color, size }) => (
             <Entypo name="home" size={size} color={color} />
           ),
-          tabBarActiveTintColor: '#5050d2',
+          tabBarActiveTintColor: PRIMARY_COLOR,
         })}
       />
-      <Tab.Screen
-        name="SharedTab"
-        component={SharedStackNavigator}
-        options={{
-          tabBarLabel: t('Shared Items')!,
-          tabBarIcon: ({ color, size }) => (
-            <MaterialIcons name="folder-shared" size={size} color={color} />
-          ),
-          tabBarActiveTintColor: '#5050d2',
-        }}
-      />
+      {currentMember ? (
+        <>
+          <Tab.Screen
+            name="SharedTab"
+            component={SharedStackNavigator}
+            options={{
+              tabBarLabel: t('Shared Items')!,
+              tabBarIcon: ({ color, size }) => (
+                <MaterialIcons name="folder-shared" size={size} color={color} />
+              ),
+              tabBarActiveTintColor: PRIMARY_COLOR,
+            }}
+          />
+          <Tab.Screen
+            name="SignInTab"
+            component={ProfileStackNavigator}
+            options={{
+              tabBarLabel: t('Profile')!,
+              tabBarIcon: ({ color, size }) => (
+                <MaterialIcons
+                  name="account-circle"
+                  size={size}
+                  color={color}
+                />
+              ),
+              tabBarActiveTintColor: PRIMARY_COLOR,
+            }}
+          />
+        </>
+      ) : (
+        <Tab.Screen
+          name="SignInTab"
+          navigationKey="SignIn"
+          // component is not used as we navigate to higher stack for signin
+          // it avoids showing tabs in the signin screen
+          component={ProfileStackNavigator}
+          listeners={({ navigation, route }) => ({
+            tabPress: (e) => {
+              // Prevent default action
+              e.preventDefault();
+
+              // Do something with the `navigation` object
+              navigation.navigate('SignIn');
+            },
+          })}
+          options={{
+            tabBarLabel: t('Sign In')!,
```

Comment:
Is the `!` necessary ? the type should be able to infer that it wont be null. Maybe we need to set a fallback ?

Reply:
Upgrading the packages, done in the next PR.